### PR TITLE
18 impl iperf test

### PR DESCRIPTION
**Pull request by wandou2018**

### CODE REVIEW

File: src/protosuites/bats/bats_btp.py [hunk 1]
```diff
@@ -1,30 +1,6 @@
-from interfaces.network import INetwork
-from protosuites.proto import IProtoSuite
-from protosuites.proto_info import IProtoInfo
+from protosuites.bats.bats_protocol import BATSProtocol
 
 
-class BTP(IProtoSuite, IProtoInfo):
+class BTP(BATSProtocol):
```
Comment (penglei0) [inline src/protosuites/bats/bats_btp.py:4]: Unable to distinguish between BTP-TUN and BRTP-TUN.

File: src/run_test.py [hunk 2]
```diff
@@ -140,8 +138,9 @@ def build_network(node_config: NodeConfig, top_config: TopologyConfig):
             linear_network.reload(local_net_top)
 
         # BTP protocol
+        bats_proto_config.hosts = range(len(linear_network.get_hosts()))
```
Comment (wandou2018) [inline src/run_test.py:141]: revert

File: src/protosuites/bats/bats_brtp.py [hunk 3]
```diff
@@ -1,30 +1,6 @@
-from interfaces.network import INetwork
-from protosuites.proto import IProtoSuite
-from protosuites.proto_info import IProtoInfo
+from protosuites.bats.bats_protocol import BATSProtocol
 
 
-class BRTP(IProtoSuite, IProtoInfo):
+class BRTP(BATSProtocol):
```
Comment (penglei0) [inline src/protosuites/bats/bats_brtp.py:4]: it's decorator pattern

File: src/testsuites/test_iperf.py [hunk 4]
```diff
@@ -23,6 +23,26 @@ def pre_process(self):
...
+        time.sleep(1)
+        client.cmd('pkill -f iperf3')
+        server.cmd('pkill -f iperf3')
+
     def _run_test(self, network: INetwork, proto: IProtoInfo):
         logging.info(
             "############### Oasis IperfTest ###########")
+        hosts = network.get_hosts()
+        client =  hosts[0]
+        server = hosts[-1]
```
Comment (penglei0) [inline src/testsuites/test_iperf.py:40]: `self.config.client_host` to `self.config.server_host`

File: src/run_test.py [hunk 5]
```diff
@@ -79,19 +78,17 @@ def setup_test(test_case_yaml, network: INetwork):
                 bats_proto_config = ProtoConfig(
                     protocol_path="/root/bats/bats_protocol",
                     protocol_args="--daemon_enabled=true",
```
Comment (wandou2018) [inline src/run_test.py:80]: --tun_protocol=BRTP
if current proto is `brtp`

File: src/containernet/containernet_network.py [hunk 6]
```diff
@@ -215,18 +208,6 @@ def _setup_topology(self):
                          self.hosts[i].name())
             self.hosts[i].cmd("echo 1 > /proc/sys/net/ipv4/ip_forward")
             self.hosts[i].cmd('sysctl -p')
-            if i + 1 < self.num_of_hosts:
```
Comment (wandou2018) [inline src/containernet/containernet_network.py]:
```
            if i + 1 < self.num_of_hosts:
                bandwidth = self.net_bw_mat[i][i + 1]
                loss = str(self.net_loss_mat[i][i + 1]) + '%'
                delay = ' delay ' + str(self.net_latency_mat[i][i + 1]) + 'ms'
                loss_delay = f"{loss} {delay}"
                self.set_link_params(
                    i, self.num_of_hosts, self.hosts[i], bandwidth, loss_delay)
            else:
                bandwidth = self.net_bw_mat[i][i - 1]
                loss = str(self.net_loss_mat[i][i - 1]) + '%'
                delay = ' delay ' + str(self.net_latency_mat[i][i - 1]) + 'ms'
                loss_delay = f"{loss} {delay}"
                self.set_link_params(
                    i, self.num_of_hosts, self.hosts[i], bandwidth, loss_delay)
```

**Pull request merged**